src/com/owncloud/android/operations/UploadFileOperation.java

```diff
@@ -477,16 +477,16 @@ protected void moveTemporalOriginalFiles(File temporalFile, File originalFile, S
      * @return      'True' if the upload was delayed until WiFi connectivity is available, 'false' otherwise.
      */
     private boolean delayForWifi() {
-        boolean delayCameraUploadsPicture = (
-                isCameraUploadsPicture() &&  PreferenceManager.cameraPictureUploadViaWiFiOnly(mContext)
-        );
-        boolean delayCameraUploadsVideo = (
-                isCameraUploadsVideo() && PreferenceManager.cameraVideoUploadViaWiFiOnly(mContext)
-        );
-        return (
-                (delayCameraUploadsPicture || delayCameraUploadsVideo) &&
-                        !ConnectivityUtils.isAppConnectedViaWiFi(mContext)
-        );
+        switch(getCreatedBy()){
+            case CREATED_AS_CAMERA_UPLOAD_PICTURE:
+                return isCameraUploadsPicture() &&  PreferenceManager.cameraPictureUploadViaWiFiOnly(mContext);
```

Comment:
Think you do not need to call `isCameraUploadsPicture()` since inside that method we are checking `mCreatedBy == CREATED_AS_CAMERA_UPLOAD_PICTURE`, same as we do with `switch(getCreatedBy()) case CREATED_AS_CAMERA_UPLOAD_PICTURE`